mini-lessons/elm/readme.md

```diff
@@ -0,0 +1,71 @@
+
+Pragmatic Studio Blog:
+
+- [What is Elm? Q&A](https://pragmaticstudio.com/blog/2015/7/23/what-is-elm-qa)
+- [Is learning Elm Worth Your Time? 4 Ways to know](https://pragmaticstudio.com/blog/2015/10/22/is-learning-elm-worth-your-time)
+
+
+
+## Why Elm? (5 - 5/50)
+
+(from [elm-lang.org](elm-lang.org))
+
+1. __No runtime exceptions:__ Yes, you read that right, no runtime exceptions. Elm’s compiler is amazing at finding errors before they can impact your users. The only way to get Elm code to throw a runtime exception is by explicitly invoking[`crash`](http://package.elm-lang.org/packages/elm-lang/core/latest/Debug#crash).
+2. __Blazing fast rendering:__ The [elm-html](http://elm-lang.org/blog/blazing-fast-html) library outperforms even React.js in [TodoMVC benchmarks](http://evancz.github.io/todomvc-perf-comparison/), and it is super simple to optimize your code by sprinkling in some [`lazy`](http://package.elm-lang.org/packages/evancz/elm-html/latest/Html-Lazy) rendering.
+3. __Libraries with guarantees:__ Semantic versioning is automatically enforced for all [community libraries](http://package.elm-lang.org/). Elm's package manager [detects any API changes](https://twitter.com/czaplic/status/601826927838650369), so breaking API changes never sneak into patches. You can upgrade with confidence.
+4. __Clean syntax:__ No semicolons. No mandatory parentheses for function calls. Everything is an expression. For even more concise code there’s also destructuring assignment, pattern matching, automatic currying, and more.
+5. __Smooth JavaScript Interop:__ No need to reinvent the wheel when there’s a JavaScript library that already does what you need. Thanks to Elm’s simple [ports](http://elm-lang.org/guide/interop) system, your Elm code can communicate with JavaScript without sacrificing guarantees.
+6. __Time-traveling debugger:__ What if you could pause time and replay all recent user inputs? What if you could make a code change and watch the results replay without a page refresh? [Try it out](http://elm-lang.org/blog/time-travel-made-easy) and see for yourself!
+
+## Getting Started with Elm (5 - 10/50)
+
+### Install
+
+#### as a module
+
+``` 
+$ npm install -g elm
+```
+
+#### … or with the [installer](http://elm-lang.org/install)
+
+If you have any problem installing with node. You can use the installer which also makes everything super easy.
+
+#### Check everything installed
+
+Run `elm` from the command line and you should see output that looks like:
+
+``` 
+Elm Platform 0.16.0 - a way to run all Elm tools
+
+Usage: elm <command> [<args>]
+
+Available commands include:
+...
+```
+
+Make sure you have version 0.16.0 — the update was about two weeks ago and some syntax we will be talking about today has changed.
+
+### Elm REPL (5 - 15/50)
+
+Pair up and look at [From JavaScript?](http://elm-lang.org/docs/from-javascript) in the Elm docs. Play around with some simple operations in the Repl. What are notable differences
```

Comment:
This may be tricky to do in 5. In my quick playing with this page + the elm repl, there are a few confusing things.

1. Immutable data?
2. `String.length "abc"` fails with `The qualifier `String` is not in scope.`
3. Unclear how to store functions in vars and call them ( `a = \x y -> x + y`, `a 3 4` returns 7)